packages/core/src/new-api/internal/journal/utils/log.ts

```diff
@@ -0,0 +1,212 @@
+    return console.log(
+      `on-chain deploy contract success - id: '${message.futureId}' - txId: '${message.txId}'`
+    );
+  }
+
+  if (isOnchainCallFunctionSuccessMessage(message)) {
+    return console.log(
+      `on-chain call function success - id: '${message.futureId}' - txId: '${message.txId}'`
+    );
+  }
+
+  if (isOnchainStaticCallSuccessMessage(message)) {
+    return console.log(
+      `on-chain static call success - id: '${
+        message.futureId
+      }' - result: '${solidityParamToString(message.result)}'`
+    );
+  }
+
+  if (isOnchainReadEventArgumentSuccessMessage(message)) {
+    return console.log(
+      `on-chain read event argument success - id: '${
+        message.futureId
+      }' - result: '${solidityParamToString(message.result)}'`
+    );
+  }
+
+  if (isOnchainSendDataSuccessMessage(message)) {
+    return console.log(
+      `on-chain send data success - id: '${message.futureId}' - txId: '${message.txId}'`
+    );
+  }
+
+  if (isOnchainContractAtSuccessMessage(message)) {
+    return console.log(
+      `on-chain contract at success - id: '${message.futureId}'`
+    );
+  }
+
+  /* execution success messages */
+
+  if (isDeployedContractExecutionSuccess(message)) {
+    return console.log(
+      `deploy contract execution success - id: '${message.futureId}' - txId: '${message.txId}'`
+    );
+  }
+
+  if (isCalledFunctionExecutionSuccess(message)) {
+    return console.log(
+      `call function execution success - id: '${message.futureId}' - txId: '${message.txId}'`
+    );
+  }
+
+  if (isStaticCallExecutionSuccess(message)) {
+    return console.log(
+      `static call execution success - id: '${
+        message.futureId
+      }' - result: '${solidityParamToString(message.result)}'`
+    );
+  }
+
+  if (isReadEventArgumentExecutionSuccess(message)) {
+    return console.log(
+      `read event argument execution success - id: '${
+        message.futureId
+      }' - result: '${solidityParamToString(message.result)}'`
+    );
+  }
+
+  if (isSendDataExecutionSuccess(message)) {
+    return console.log(
+      `send data execution success - id: '${message.futureId}' - txId: '${message.txId}'`
+    );
+  }
+
+  if (isContractAtExecutionSuccess(message)) {
+    return console.log(
+      `contract at execution success - id: '${message.futureId}'`
+    );
+  }
+
+  /* hold & failure messages */
+
+  if (isExecutionHold(message)) {
+    return console.log(`Execution on hold`);
+  }
+
+  if (isOnChainFailureMessage(message)) {
+    return console.log(
+      `on chain failure - futureId: '${message.futureId}' error: '${message.error.message}'`
+    );
+  }
+
+  if (isExecutionFailure(message)) {
+    return console.log(
+      `execution failure - futureId: '${message.futureId}' error: '${message.error.message}'`
+    );
+  }
+
+  if (isWipeMessage(message)) {
+    return console.log(`wiping journal`);
+  }
+
+  throw new Error("Unknown journal message");
```

Comment:
Lets wrap this in an assertNever() style function, to get a compile time check when we add new Journalable messages.